File: OC/Navigation_With_TableView/Navigation_With_TableView/ViewController.m
//
//  ViewController.m
//  Navigation_With_TableView
//
//  Created by ligulfzhou on 1/28/16.
//  Copyright © 2016 ligulfzhou. All rights reserved.
//

#import "ViewController.h"
#import "SecondViewController.h"

@interface ViewController ()

@property (nonatomic, strong) NSArray *array;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"view controller";
    
    _array = @[@"label1", @"label2", @"label3", @"label4", @"label5"];
    
    _tableView = [[UITableView alloc] initWithFrame:self.view.bounds];
    _tableView.dataSource = self;
    _tableView.delegate = self;
    [self.view addSubview:_tableView];
    
    if (YES) {
        
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark
#pragma mark - UItableviewDataSource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [_array count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"cell"];
    }
    cell.textLabel.text = _array[indexPath.row];
    return cell;
}

#pragma mark
#pragma mark - UITableViewDelegate
-(void) tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    SecondViewController *vc = [[SecondViewController alloc] init];
    vc.label = _array[indexPath.row];
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: Swift/UIScrollView/UIScrollView/ViewController.swift
//
//  ViewController.swift
//  UIScrollView
//
//  Created by ligulfzhou on 2/19/16.
//  Copyright © 2016 ligulfzhou. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {

    var tableview:UITableView!
    let identifier:String = "TableCellIdentifier"
    let tableRows = ["image Scrolll", "custom view scroll", "paged", "paged with peeking"]
    let viewControllers = [ImageScrollViewController.classForCoder()]
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "scroll view"
        
        tableview = UITableView(frame: view.bounds)
        tableview.registerClass(UITableViewCell.classForCoder(), forCellReuseIdentifier: identifier)
        tableview.dataSource = self
        tableview.delegate = self
        view.addSubview(tableview)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    //MARK: uitableview data source
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return tableRows.count
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier(identifier, forIndexPath: indexPath)
        cell.textLabel?.text = tableRows[indexPath.row]
        return cell
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
//        let viewController = viewControllers[indexPath.row]() as UIViewController
        
        let viewController = ImageScrollViewController()
        navigationController?.pushViewController(viewController, animated: true)
    }
}

